pkg/cluster/clustermsi_test.go

```diff
@@ -308,3 +311,165 @@ func TestClusterMsiSecretName(t *testing.T) {
 		})
 	}
 }
+
+func TestClusterIdentityIDs(t *testing.T) {
+	ctx := context.Background()
+
+	mockGuid := "00000000-0000-0000-0000-000000000000"
+	clusterRGName := "aro-cluster"
+	clusterName := "aro-cluster"
+
+	clusterResourceId := strings.ToLower(fmt.Sprintf("/subscriptions/%s/resourceGroups/%s/providers/Microsoft.RedHatOpenShift/openShiftClusters/%s", mockGuid, clusterRGName, clusterName))
+
+	miName := "aro-cluster-msi"
+	miResourceId := fmt.Sprintf("/subscriptions/%s/resourceGroups/%s/providers/Microsoft.ManagedIdentity/userAssignedIdentities/%s", mockGuid, clusterRGName, miName)
+
+	msiDataPlaneNotFoundError := `failed to get credentials: Request information not available
+--------------------------------------------------------------------------------
+RESPONSE 404: 
+ERROR CODE UNAVAILABLE
+--------------------------------------------------------------------------------
+Response contained no body
+--------------------------------------------------------------------------------
+`
+	miClientId := "ffffffff-ffff-ffff-ffff-ffffffffffff"
+	miObjectId := "99999999-9999-9999-9999-999999999999"
+	placeholderString := "placeholder"
+
+	for _, tt := range []struct {
+		name             string
+		doc              *api.OpenShiftClusterDocument
+		msiDataplaneStub policy.ClientOptions
+		wantDoc          *api.OpenShiftClusterDocument
+		wantErr          string
+	}{
```

Comment:
Are we missing the test cases for ExplicitIdentities checks?

Reply:
Unfortunately we cannot add unit tests for when the ExplicitIdentities are missing, because that will cause the client instantiation itself to return the `msiDataPlaneNotFound` error rather than invoking/testing our logic and custom errors.